### PR TITLE
[feature](table-metadata) support altering the property "light_schema_change" for the tables which created before 1.2

### LINKED ISSUE #17094: [Enhancement] Support altering the property "light_schema_change" for the tables which created before 1.2
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

There is similar issue [related-issue](https://github.com/apache/doris/issues/15601) but it's closed. 
For the tables which created before 1.2, when the BE upgrade to  1.2, the user cann't alter the property "light_schema_change". 

### Solution

The different whether light_schema_change or not is that the column ids are set in FE or BE. Before 1.2, the column ids are set in BE. If light_schema_change = true, the column ids are set in FE. So when we want to alter a non-light-schema-change table to support light-schema-change, we need to get the column ids from BE and rewrite the FE table meta. The feature is important very much because when a table altered to light-schema-change and it will not support to rollback. It mean that we need plenty of patience to test and familiar with FE/BE metadata code.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Hastyshell**
# Proposed changes

Issue Number: close #17094 

## Problem summary

Try to:
1. Enable the stmt of altering light_schema_change table property and add some pre-check.
2. Add rpc method to read column uniqe ids from BE.
3. Read column ids from BE and rerwrite the meta of both base indexs and rollup indexs for target old olap table.
4. Write corresponding edit log

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments
Related docs and tests will be added later.

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -1845,6 +1868,61 @@ public int getAsInt() {
...
+        final List<TFetchColIdsEntry> resultList = response.getResultList();
+
+        // write meta
+        resultList.forEach(entry -> {
+            final long tabletId = entry.getTabletId();
+            final List<Integer> colIds = entry.getColIds();
+            final MaterializedIndex index = tabletIdToIdx.get(tabletId);
+            final MaterializedIndexMeta indexMeta = olapTable.getIndexMetaByIndexId(index.getId());
+            final List<Column> columns = indexMeta.getSchema();
+            // TODO: think about can we ensure the order consistency of columns?
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:1915]: I think you could not assume that. Should use column name to column unique id mapping.

Comment (TangSiyang2001) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:1915]: Thank you for resolving what I've concerned.

**Comment by yiguolei** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run clickbench

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.07 seconds
 stream load tsv:          450 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230319150724_clickbench_pr_116899.html

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run feut

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run feut

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run beut

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run arm

**Comment by TangSiyang2001** [comment]
run p1

**Comment by dataroaring** [comment]
What if two adding colums op running concurrently. e.g. op1 is running in heavy mode and we modify table to enable light schema change then op2 runs before op1 finishes.

**Comment by dataroaring** [comment]
We should check all tablets's columns's unique id in be, It seems that they should be same, but we should check it.

**Comment by TangSiyang2001** [comment]
> What if two adding colums op running concurrently. e.g. op1 is running in heavy mode and we modify table to enable light schema change then op2 runs before op1 finishes.

According to the compatibility matrix, op MODIFY_TABLE_PROPERTY will confict with op SCHEMA_CHANGE, so we can avoid this situation.

**Comment by TangSiyang2001** [comment]
> We should check all tablets's columns's unique id in be, It seems that they should be same, but we should check it.

Does it mean that we should collect all tablets for a materialized index and check whether their column ids are consistent?

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run arm

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run arm

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run arm

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**